channel/consensus_channel/consensus_channel.go

```diff
@@ -143,38 +143,26 @@ var ErrInvalidDeposit = fmt.Errorf("unable to divert to guarantee: invalid depos
 var ErrInsufficientFunds = fmt.Errorf("unable to divert to guarantee: insufficient funds")
 var ErrDuplicateGuarantee = fmt.Errorf("duplicate guarantee detected")
 
-// DivertToGuarantee deducts g.amount from o.left's balance, and
-// adds g to o.guarantees
-func (o LedgerOutcome) DivertToGuarantee(p Add) (LedgerOutcome, error) {
-	g := p.Guarantee
-
-	if types.Gt(&p.LeftDeposit, &g.amount) {
-		return LedgerOutcome{}, ErrInvalidDeposit
-	}
-
-	if types.Gt(&g.amount, &o.left.amount) {
-		return LedgerOutcome{}, ErrInsufficientFunds
-	}
-
-	o.left.amount.Sub(&o.left.amount, &p.LeftDeposit)
-	rightDeposit := p.RightDeposit()
-	o.right.amount.Sub(&o.right.amount, &rightDeposit)
-
-	_, found := o.guarantees[g.target]
-	if found {
-		return LedgerOutcome{}, ErrDuplicateGuarantee
-	}
-	o.guarantees[g.target] = g
-
-	return o, nil
-}
-
 // Vars stores the turn number and outcome for a state in a consensus channel
 type Vars struct {
 	TurnNum uint64
 	Outcome LedgerOutcome
 }
 
+// clone returns a deep clone of v
+func (v Vars) clone() Vars {
+	v.Outcome.left.amount = *new(big.Int).Set(&v.Outcome.left.amount)
+	v.Outcome.right.amount = *new(big.Int).Set(&v.Outcome.right.amount)
+
+	guarantees := make(map[types.Destination]Guarantee)
+	for d, g := range v.Outcome.guarantees {
+		guarantees[d] = g
```

Comment:
`Guarantee` struct has an `amount` field that is a `big.Int`. It seems that this `big.Int` is treated differently than the `Outcome.left.amount` and `Outcome.right.amount` at the beginning of the function. On this line, we use assignment. For outcome amounts, we allocate a new `big.Int` struct.

Reply:
It seems that the first 2 lines of this function might be unnecessary?

Reply:
You're definitely right, and they should not be treated differently.

> It seems that the first 2 lines of this function might be unnecessary?

I learned that they are actually necessary! See https://github.com/statechannels/go-nitro/pull/411/commits/4612ae86e3935197c15f500a708dab0d42a612a1

This is fixed in cc57d4e3c6a793ab5a045b5596bae908a613ce61

Reply:
Good to know that a `big.Int` struct contains a slice!